tests/Control/SMIMEMailerTest.php

```diff
@@ -0,0 +1,243 @@
+    }
+
+    public function testSigningErrorWithIncorrectSigningPassword()
+    {
+        $recipientSigningCrt = $this->getAsset('/assets/smime_test_sender.crt');
+        $recipientSigningKey = $this->getAsset('/assets/smime_test_sender.key');
+        $email = $this->buildEmail(
+            'Email smime signed by sender',
+            '<h1>This is not a confidential email.</h1>'
+        );
+
+        $partialMockMailer = $this->createPartialMockedMIMEMailer([
+            null,
+            $recipientSigningCrt,
+            $recipientSigningKey,
+            'IncorrectPassword!'
+        ]);
+
+        $partialMockMailer->expects($this->once())->method('sendSwift')->with(
+            self::callback(function ($message) {
+                $exception = null;
+                try {
+                    // Calling toString encrypts and signs.
+                    // Here we expect an exception because of incorrect password
+                    $message->toString();
+                } catch (Exception $e) {
+                    $exception = $e;
+                } finally {
+                    self::assertEquals(
+                        'openssl_pkcs7_sign(): Error getting private key',
+                        $exception->getMessage()
+                    );
+                }
+
+                return true;
+            })
+        );
+
+        $partialMockMailer->send($email);
+    }
+
+    public function testSigningAndEncryption()
+    {
+        $recipientEncryptionCrt = $this->getAsset('/assets/smime_test_recipient.crt');
+        $recipientSigningCrt = $this->getAsset('/assets/smime_test_sender.crt');
+        $recipientSigningKey = $this->getAsset('/assets/smime_test_sender.key');
+        $email = $this->buildEmail(
+            'Email encrypted and smime signed by sender',
+            '<h1>This is a confidential email.</h1>'
+        );
+
+        $partialMockMailer = $this->createPartialMockedMIMEMailer([
+            $recipientEncryptionCrt,
+            $recipientSigningCrt,
+            $recipientSigningKey,
+            'Test123!'
+        ]);
+
+        $partialMockMailer->expects($this->once())->method('sendSwift')->with(
+            self::callback(function ($message) {
+                self::assertStringNotContainsString('This is a confidential email.', $message->toString());
+
+                // Note, messages are signed and then encrypted, so the signing particulars are not seen unencrypted
+                self::assertStringNotContainsString(
+                    'Content-Type: application/x-pkcs7-signature; name="smime.p7s"',
+                    $message
+                );
+
+                self::assertStringNotContainsString('This is an S/MIME signed message', $message);
+
+                return true;
+            })
+        );
+
+        $partialMockMailer->send($email);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        // Disable as we don't have any db to reset
+    }
```

Comment:
Does this need to be here? If so, maybe add a docblock to explain why...

Reply:
Kinda not needed if running tests that has a database configured, but I was running the tests without and this stopped an error from occurring. I'm guessing no harm leaving in since the tests don't make use of a database, so have just added a further docblock comment for explanation.